core/src/test/scala/unit/kafka/metrics/MetricsTest.scala

```diff
@@ -121,6 +125,17 @@ class MetricsTest extends KafkaServerTestHarness with Logging {
     // Produce a few messages to create the metrics
     TestUtils.produceMessages(servers, topic, nMessages)
 
+    // Check the log size for each broker so that we can distinguish between failures caused by replication issues
+    // versus failures caused by the metrics
+    val topicPartition = new TopicPartition(topic, 0)
+    servers.foreach { server =>
+      val log = server.logManager.logsByTopicPartition.get(new TopicPartition(topic, 0))
+      val brokerId = server.config.brokerId
+      val logSize = log.map(_.size)
+      assertTrue(s"Expected broker $brokerId to have a Log for $topicPartition with positive size, actual: $logSize",
+        logSize.map(_ > 0).getOrElse(false))
+    }
+
```

Comment:
@ijuma I may be missing something, but isn't `replicationBytesIn` referring to a broker-level metric rather than a topic metric? Looking at the failure https://builds.apache.org/job/kafka-trunk-jdk7/2190/ for instance, the error says `Unable to find metric ReplicationBytesInPerSec`. This metric should have been created when the broker started up?